server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeAttribute.java

```diff
@@ -144,62 +171,117 @@ private RepositoryMetadata buildRepositoryMetadata(DiscoveryNode node, String na
+            if (attributes.containsKey(key)) {
+                return new Tuple<>(attributes.get(key), key);
+            }
         }
-        if (node.getAttributes().containsKey(REMOTE_STORE_ROUTING_TABLE_REPOSITORY_NAME_ATTRIBUTE_KEY)) {
-            repositoryNames.add(validateAttributeNonNull(node, REMOTE_STORE_ROUTING_TABLE_REPOSITORY_NAME_ATTRIBUTE_KEY));
+        return null;
+    }
+
+    private Map<String, String> getValidatedRepositoryNames(DiscoveryNode node) {
+        Set<Tuple<String, String>> repositoryNames = new HashSet<>();
+        if (containsKey(node.getAttributes(), REMOTE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEYS)
+            || containsKey(node.getAttributes(), REMOTE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEYS)) {
+            repositoryNames.add(validateAttributeNonNull(node, REMOTE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEYS));
+            repositoryNames.add(validateAttributeNonNull(node, REMOTE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEYS));
+            repositoryNames.add(validateAttributeNonNull(node, REMOTE_CLUSTER_STATE_REPOSITORY_NAME_ATTRIBUTE_KEYS));
+        } else if (containsKey(node.getAttributes(), REMOTE_CLUSTER_STATE_REPOSITORY_NAME_ATTRIBUTE_KEYS)) {
+            repositoryNames.add(validateAttributeNonNull(node, REMOTE_CLUSTER_STATE_REPOSITORY_NAME_ATTRIBUTE_KEYS));
+        }
+        if (containsKey(node.getAttributes(), REMOTE_ROUTING_TABLE_REPOSITORY_NAME_ATTRIBUTE_KEYS)) {
+            repositoryNames.add(validateAttributeNonNull(node, REMOTE_ROUTING_TABLE_REPOSITORY_NAME_ATTRIBUTE_KEYS));
         }
 
-        return repositoryNames;
+        Map<String, String> repoNamesWithPrefix = new HashMap<>();
+        repositoryNames.forEach(t -> {
+            String[] attrKeyParts = t.v2().split("\\.");
+            repoNamesWithPrefix.put(t.v1(), attrKeyParts[0]);
+        });
+
+        return repoNamesWithPrefix;
     }
 
     public static boolean isRemoteStoreAttributePresent(Settings settings) {
-        return settings.getByPrefix(Node.NODE_ATTRIBUTES.getKey() + REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX).isEmpty() == false;
+        for (String prefix : REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX) {
+            if (settings.getByPrefix(Node.NODE_ATTRIBUTES.getKey() + prefix).isEmpty() == false) {
+                return true;
+            }
+        }
+        return false;
     }
 
     public static boolean isRemoteDataAttributePresent(Settings settings) {
-        return settings.getByPrefix(Node.NODE_ATTRIBUTES.getKey() + REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY).isEmpty() == false
-            || settings.getByPrefix(Node.NODE_ATTRIBUTES.getKey() + REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY).isEmpty() == false;
+        return isSegmentRepoConfigured(settings) || isTranslogRepoConfigured(settings);
+    }
+
+    public static boolean isSegmentRepoConfigured(Settings settings) {
+        for (String prefix : REMOTE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEYS) {
```

Comment:
would it throw an exception in case repo is configured with both prefixes on single node?

Reply:
No, this would pick the first prefix (in the list order) it encounters and then pick the repository settings from the corresponding prefix.

Reply:
Ideally we shouldnt allow repos with both the prefix?

Reply:
We can probably fail the while loading this settings instead of ignoring the 2nd prefix. I have create a follow-up issue (https://github.com/opensearch-project/OpenSearch/issues/16605) to fix this on a seperate PR as this code path is invoked from multiple flows (boot-up , node-join and index create).